src/heartbeat.ts

```diff
@@ -0,0 +1,95 @@
+import * as vscode from 'vscode';
+import { Disposable } from './common/dispose';
+import Log from './common/logger';
+import { withServerApi } from './internalApi';
+
+export class HeartbeatManager extends Disposable {
+
+    private lastActivity = new Date().getTime();
+    private gitpodHost = vscode.workspace.getConfiguration('gitpod').get<string>('host')!;
```

Comment:
Should not we pass it via contructor to be sure? I would prefer that avoid using this setting, but provide value if we have it from params or other sources.
User can change this value for some reasons and then it will break us in weird ways. It would be enough to content the logic of checking and updating a setting in one place like `getGitpodSession` function.